tests/test_pydeseq2.py

```diff
@@ -35,18 +35,28 @@ def metadata():
     )
 
 
-@pytest.mark.parametrize("independendent_filtering", [True, False])
-def test_deseq(counts_df, metadata, independendent_filtering, tol=0.02):
+@pytest.mark.parametrize(
+    "independent_filtering, trend_fit_type",
+    [(True, "parametric"), (True, "mean"), (False, "parametric")],
+)
+def test_deseq(counts_df, metadata, independent_filtering, trend_fit_type, tol=0.02):
```

Comment:
Don't ask me how I ended up here, but still giving my advice. Generally speaking, it's better to isolate tests that serve a different purpose. Hints here that you are not doing that is the presence of if/else statements in the test based on `@pytest.mark.parametrize`'s inputs. Usually, what you want to have in there are inputs and expected outputs to provide exhaustivity (but the actual test will not change), eg:
```py
import pytest 

def add(a, b):
    return a + b

@pytest.mark.parametrize("a, b, expected", [
    (1, 2, 3),  # Positive integers
    (-1, -1, -2),  # Negative integers
    (0, 0, 0),  # Zeros
    (3.5, 2.5, 6.0),  # Floating point numbers
    ("Hello, ", "world!", "Hello, world!")  # Strings
])
def test_add(a, b, expected):
    assert add(a, b) == expected
```

---

What I would suggest here is to split your tests:
```py
def test_deseq_independent_filtering_parametric_fit(counts_df, metadata, tol=0.02):
    test_path = str(Path(os.path.realpath(tests.__file__)).parent.resolve())
    r_res = pd.read_csv(
        os.path.join(test_path, "data/single_factor/r_test_res.csv"),
        index_col=0,
    )
    dds = DeseqDataSet(
        counts=counts_df,
        metadata=metadata,
        design_factors="condition",
        trend_fit_type="parametric",
    )
    dds.deseq2()
    res = DeseqStats(dds, independent_filter=True)
    ...
  

def test_deseq_independent_filtering_mean_fit(counts_df, metadata, tol=0.02):
    ...


def test_deseq_without_independent_filtering_parametric_fit(counts_df, metadata, tol=0.02):
    ...
```

I know that:
* It makes the test names way more verbose (but is that really a bad thing?)
* You will have to repeat the assertion part in every function and I'm sure you've been told many times that code duplication is bad, but other methods exist to prevent from that

Feel free to do whatever you want with this comment!